### PR TITLE
Adds option for a Transaction's flags to be a List of ints

### PR DESCRIPTION
**Pull request by mvadari**
## High Level Overview of Change

Title says it all.

### Context of Change

Python library 1.1.0

### Type of Change

- [x] New feature (non-breaking change which adds functionality)
- [x] Tests (You added tests for code that already exists, or your new feature included in this PR)

## Test Plan

Added simple test. CI passes.

### CODE REVIEW

File: xrpl/models/transactions/transaction.py [hunk 1]
```diff
@@ -209,7 +209,15 @@ def to_dict(self: Transaction) -> Dict[str, Any]:
...
-        return {**super().to_dict(), "transaction_type": self.transaction_type.value}
+        dictionary = {
+            **super().to_dict(),
+            "transaction_type": self.transaction_type.value,
+        }
+        if isinstance(self.flags, list):
+            dictionary["flags"] = 0
+            for flag in self.flags:
+                dictionary["flags"] |= flag
+        return dictionary
```
Comment (ledhed2222) [inline xrpl/models/transactions/transaction.py:220]: suggestion:

```python
def to_dict(self: Transaction) -> Dict[str, Any]:
  return {**super().to_dict(), "transaction_type": self.transaction_type.value, "flags": self._flags_to_dict()}
  
def _flags_to_dict(self: Transaction) -> int:
  if isinstance(self.flags, int):
    return self.flags
  accumulator = 0
  for flag in self.flags:
    accumulator |= flag
  return accumulator
```

factoring out into helper - just readability imo. take it or leave it

File: xrpl/models/transactions/transaction.py [hunk 2]
```diff
@@ -256,7 +264,10 @@ def has_flag(self: Transaction, flag: int) -> bool:
         Returns:
             Whether the transaction has the given flag value set.
         """
-        return self.flags & flag != 0
+        if isinstance(self.flags, int):
+            return self.flags & flag != 0
+        else:  # is List[int]
+            return flag in self.flags
```
Comment (ledhed2222) [inline xrpl/models/transactions/transaction.py:270]: don't need the `else` here. could also be expressed as:

```python
return (isinstance(self.flags, int) and self.flags & flag != 0) or flag in self.flags
```

**Review by ledhed2222 — APPROVED** [review]

**Comment by ledhed2222** [comment]
changelog!

File: xrpl/models/transactions/transaction.py (hunk 2 above)
Comment (mvadari) [reply xrpl/models/transactions/transaction.py:270]: I think that's messier to read

**Pull request merged**